File: receiver.cpp
#include <cstdlib>
#include <iostream>
#include <sstream>
#include <string>
#include <unistd.h>
#include <RF24/RF24.h>

#include <ctime>
#include <fstream>

using namespace std;
RF24 radio(22,0);

/********************************/

// Radio pipe addresses for the 2 nodes to communicate.
const uint64_t pipes[2] = {0xF0F0F0F0E1LL, 0x7365727631LL};

    struct secureMessage{
    char message [16];
  };

void writeToFile (char input[16]) {
    // current date/time based on current system
    time_t now = time(0);
    // Hängt Sachen an die Datei an
    std::ofstream data("datafifo", std::ios_base::app | std::ios_base::out);
    
    data << now << " " << input << " " <<"\n";
}
  
int main(int argc, char** argv){

  cout << "\n RF24 Receiver for Studienarbeit\n";

  
  radio.begin();
  radio.setPayloadSize(sizeof(secureMessage));  //Groesse der gesendeten Daten
  radio.setAutoAck(1); 
  radio.setDataRate(RF24_250KBPS); //250kbs
  radio.setPALevel(RF24_PA_MAX);
  radio.setChannel(90);
  radio.setRetries(15,15);
  radio.setCRCLength(RF24_CRC_16);
  radio.openWritingPipe(pipes[1]);
  radio.openReadingPipe(1,pipes[0]);

  
  //radio.printDetails();



    
	
	radio.startListening();
	
	// forever loop
	while (1)
	{
        secureMessage t_message;

            if( radio.available()){                             // Variable for the received timestamp
                while (radio.available()) {                     // While there is data ready
                radio.read( &t_message, sizeof(t_message) ); 
                writeToFile(t_message.message);
                
                }
            }
    } // forever loop

  return 0;
}

                     

File: receiver.py
#!/usr/bin/env python

#from __future__ import print_function
import time
from RF24 import *
import pdb
import RPi.GPIO as GPIO
import os, sys

irq_gpio_pin = None



radio = RF24(22, 0);
pipes = [0xF0F0F0F0E1, 0x7365727631]

def setup():
  radio.begin();
  radio.payloadSize = 16        # Hardcoded size TODO --> len(message)
  #radio.enableDynamicPayloads()
  radio.setAutoAck(1); 
  radio.setDataRate(RF24_250KBPS); #250kbs
  radio.setPALevel(RF24_PA_MAX);
  radio.setChannel(90);
  radio.setRetries(15,15);
  radio.setCRCLength(RF24_CRC_16);
  radio.openWritingPipe(pipes[1]);
  radio.openReadingPipe(1,pipes[0]);
  #radio.maskIRQ(0,0,1);
  radio.startListening();
  
def writeToFile(received_payload):
	print('Got payload size={} value="{}"'.format(radio.payloadSize, received_payload.decode('utf-8')))
	
	#Timestamp erzeugen
	utime=int(time.time())
	
	# Hier aktuelle Zeit einfuegen
	data.write("1481278800" + " " + received_payload + " \n") #% utime
	data.flush()
	
def receive():
    if radio.available():
		while radio.available():
			len = radio.payloadSize

			receive_payload = radio.read(len)
			writeToFile(receive_payload)	
        

#Hauptprogramm:			
print ("Hallo")
pdb.set_trace()
setup()
radio.printDetails()
data = open('datafifo', 'a+')
print("Hauptprogramm:")
while 1:
	receive()
	
# spaeter durch interrupt ersetzten
	time.sleep(1)
data.close()
